docs/tutorials/python/profiler.md

```diff
@@ -0,0 +1,198 @@
+                                   batch_size=64, shuffle=True)
+```
+
+Let's define a method that will run one training iteration given data and label.
+
+```python
+# Use GPU is available
+ctx = mx.gpu() if mx.test_utils.list_gpus() else mx.cpu()
+
+# Initialize the parameters with random weights
+net.collect_params().initialize(mx.init.Xavier(), ctx=ctx)
+
+# Use SGD optimizer
+trainer = gluon.Trainer(net.collect_params(), 'sgd', {'learning_rate': .1})
+
+# Softmax Cross Entropy is a frequently used loss function for multi-classs classification
+softmax_cross_entropy = gluon.loss.SoftmaxCrossEntropyLoss()
+
+# A helper function to run one training iteration
+def run_training_iteration(data, label):
+    
+    # Load data and label is the right context
+    data = data.as_in_context(ctx)
+    label = label.as_in_context(ctx)
+    
+    # Run the forward pass
+    with autograd.record():
+        output = net(data)
+        loss = softmax_cross_entropy(output, label)
+    
+    # Run the backward pass
+    loss.backward()
+    
+    # Apply changes to parameters
+    trainer.step(data.shape[0])
+```
+
+### Starting and stopping the profiler
+
+When the first forward pass is run on a network, MXNet does a number of housekeeping tasks including inferring the shapes of various parameters, allocating memory for intermediate and final outputs, etc. For these reasons, profiling the first iteration doesn't provide accurate results. We will, therefore skip the first iteration.
+
+```python
+# Run the first iteration without profiling
+itr = iter(train_data)
+run_training_iteration(*next(itr))
+```
+
+We'll run the next iteration with the profiler turned on.
+
+```python
+data, label = next(itr)
+
+# Ask the profiler to start recording
+profiler.set_state('run')
+
+run_training_iteration(*next(itr))
+
+# Ask the profiler to stop recording
+profiler.set_state('stop')
+```
+
+Between running and stopping the profiler, you can also pause and resume the profiler using `profiler.pause()` and `profiler.resume()` respectively to profile only parts of the code you want to profile.
+
+### Viewing profiler output
+
+There are two ways to view the information collected by the profiler. You can either view it in the console or you can view a more graphical version in a browser.
+
+#### 1. View in console
+
+You can use the `profiler.dumps()` method to view the information collected by the profiler in the console. The collected information contains time taken by each operator, time taken by each C API and memory consumed in both CPU and GPU.
+
+```python
+print(profiler.dumps())
+```
+
+![Profile Statistics](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/tutorials/python/profiler/profile_stats.png)<!--notebook-skip-line-->
+
+#### 2. View in browser
+
+You can also dump the information collected by the profiler into a `json` file using the `profiler.dump()` function and view it in a browser.
+
+```python
+profiler.dump()
```

Comment:
So, the difference between getting a plain text version vs. json is in calling "dump**s**()" vs "dump()"? Is it possible to change this signature?

Reply:
I guess the `s` in `dumps` indicates the method returns a string. Like `pickle.dumps()` or `json.dumps()`